napari_cellseg3d/code_models/model_instance_seg.py

```diff
@@ -1,24 +1,83 @@
-from __future__ import division
-from __future__ import print_function
-
 from dataclasses import dataclass
 from typing import List
 
 import numpy as np
-
-# from skimage.measure import marching_cubes
-# from skimage.measure import mesh_surface_area
-from skimage.measure import label
-from skimage.measure import regionprops
+import pyclesperanto_prototype as cle
+from qtpy.QtWidgets import QWidget
+from skimage.measure import label, regionprops
 from skimage.morphology import remove_small_objects
 from skimage.segmentation import watershed
+
+# from skimage.measure import mesh_surface_area
+# from skimage.measure import marching_cubes
 from tifffile import imread
 
-from napari_cellseg3d.utils import fill_list_in_between
-from napari_cellseg3d.utils import sphericity_axis
+from napari_cellseg3d import interface as ui
+from napari_cellseg3d.utils import LOGGER as logger
+from napari_cellseg3d.utils import fill_list_in_between, sphericity_axis
 
 # from napari_cellseg3d.utils import sphericity_volume_area
 
+# list of methods :
+WATERSHED = "Watershed"
+CONNECTED_COMP = "Connected Components"
+VORONOI_OTSU = "Voronoi-Otsu"
+
+
+class InstanceMethod:
+    def __init__(
+        self,
+        name: str,
+        function: callable,
+        num_sliders: int,
+        num_counters: int,
+        widget_parent: QWidget = None,
+    ):
+        """
+        Methods for instance segmentation
+
+        Args:
+            name: Name of the instance segmentation method (for UI)
+            function: Function to use for instance segmentation
+            num_sliders: Number of Slider UI elements needed to set the parameters of the function
+            num_counters: Number of DoubleIncrementCounter UI elements needed to set the parameters of the function
+            widget_parent: parent for the declared widgets
+
+        """
+        self.name = name
+        self.function = function
+        self.counters: List[ui.DoubleIncrementCounter] = []
+        self.sliders: List[ui.Slider] = []
+        if num_sliders > 0:
```

Comment:
Might be a tad bit cleaner if these two are in separate methods setup_sliders and setup_counters, or in one method setup_widgets(num_sliders, num_counters)

Reply:
Fixed in #36